core_chat_data/src/commonMain/kotlin/net/thechance/mena/core_chat/data/contacts/ContactsRepositoryImpl.kt

```diff
@@ -0,0 +1,39 @@
+package net.thechance.mena.core_chat.data.contacts
+
+import net.thechance.mena.core_chat.data.contacts.source.remote.ContactsRemoteDataSource
+import net.thechance.mena.core_chat.domain.entity.Contact
+import net.thechance.mena.core_chat.domain.exception.FailException
+import net.thechance.mena.core_chat.domain.repository.ContactsRepository
+
+class ContactsRepositoryImpl(
+    private val contactsRemoteDataSource: ContactsRemoteDataSource
+) : ContactsRepository {
+
+    override suspend fun getUserContacts(
+        userId: String,
+        pageNumber: Int,
+        pageSize: Int
+    ): List<Contact> {
```

Comment:
```suggestion
    ): Paged<Contact> {
```
Will be useful if have an information about total pages or total items

Reply:
what is the benefit of doing this ?
why I need to know the total pages or total items ?

Reply:
To know if I have already reached the last page or not